community/security/src/main/java/org/neo4j/server/security/auth/Credential.java

```diff
@@ -106,7 +115,9 @@ public boolean equals( Object o )
 
         Credential that = (Credential) o;
 
-        return byteEquals( this.salt, that.salt ) && byteEquals( this.passwordHash, that.passwordHash );
+        boolean saltEquals = byteEquals( this.salt, that.salt );
+        boolean passwordEquals = byteEquals( this.passwordHash, that.passwordHash );
+        return saltEquals && passwordEquals;
```

Comment:
Was this condition split for readability or to always execute both branches? Maybe add a comment if it is the later

Reply:
Yes this was done to always to execute both branches. We'll add the comment.